R/terminal.R

```diff
@@ -0,0 +1,270 @@
+
+#' Send Text to a Terminal
+#'
+#' Send text to an existing terminal.
+#'
+#' @param id The terminal id. The \code{id} is obtained from
+#'   \code{\link{terminalList}()}, \code{\link{terminalVisible}()},
+#'   \code{\link{terminalCreate}()}, or \code{\link{terminalExecute}()}.
+#' @param text Character vector containing text to be inserted.
+#'
+#' @note The \code{terminalSend} function was added in version 1.1.305 of RStudio.
+#'
+#' @examples
+#' \dontrun{
+#' rstudioapi::terminalSend(id='My Terminal', 'ls -l\n')
+#' }
+#'
+#' @export
+terminalSend <- function(id, text) {
+  callFun("terminalSend", id, text)
+}
+
+
+#' Clear Terminal Buffer
+#'
+#' Clears the buffer for specified terminal.
+#'
+#' @param id The terminal id. The \code{id} is obtained from
+#'   \code{\link{terminalList}()}, \code{\link{terminalVisible}()},
+#'   \code{\link{terminalCreate}()}, or \code{\link{terminalExecute}()}.
+#'
+#' @note The \code{terminalClear} function was added in version 1.1.305 of RStudio.
+#'
+#' @examples
+#' \dontrun{
+#' rstudioapi::terminalClear('Terminal 1')
+#' }
+#'
+#' @export
+terminalClear <- function(id) {
+  callFun("terminalClear", id)
+}
+
+
+#' Create a Terminal
+#'
+#' Create a new Terminal.
+#'
+#' @param id The desired terminal caption. When \code{NULL} or blank,
+#' the terminal caption will be chosen by the system.
+#' @param show If FALSE, terminal won't be brought to front
+#'
+#' @return The terminal identifier as a character vector (\code{NULL} if
+#'   unable to create the terminal or the given terminal caption is already
+#'   in use).
+#'
+#' @note The \code{terminalCreate} function was added in version 1.1.305 of RStudio.
+#'
+#' @examples
+#' \dontrun{
+#' terminalId <- rstudioapi::terminalCreate('My Terminal')
+#' }
+#'
+#' @export
+terminalCreate <- function(caption = NULL, show = TRUE) {
+  callFun("terminalCreate", caption, show)
+}
+
+
+#' Is Terminal Busy
+#'
+#' Are terminals reporting that they are busy?
+#'
+#' @param id The terminal id. The \code{id} is obtained from
+#'   \code{\link{terminalList}()}, \code{\link{terminalVisible}()},
+#'   \code{\link{terminalCreate}()}, or \code{\link{terminalExecute}()}.
+#'
+#' @return a boolean
+#'
+#' @note The \code{terminalBusy} function was added in version 1.1.305 of RStudio.
+#'
+#' @export
```

Comment:
A small example here might be nice too -- would it make sense to wait on a running terminal process with something like:

```
# wait until a busy terminal is finished
while (terminalBusy(id)) {
    Sys.sleep(0.1)
}
```

?